test/spec/email/email.spec.ts

```diff
@@ -0,0 +1,180 @@
+/* eslint-disable @typescript-eslint/no-unsafe-call */
+import { expect } from 'chai';
+import sinon from 'sinon';
+import CommunicationServiceManager from '../../../src/modules/communication/communication-service-manager';
+import EmailService from '../../../src/modules/communication/email-service';
+import chai from 'chai';
+import chaiAsPromised from 'chai-as-promised';
```

Comment:
We can also use async/await and `assert.throws` from chai. Can keep it if it's easy to use.

Reply:
@akashlakhera  sir, I found this an easy and readable way for test cases therefore I used it.